tests/components/history/test_websocket_api.py

```diff
@@ -1584,3 +1584,233 @@ async def test_overflow_queue(
     assert listeners_without_writes(
         hass.bus.async_listeners()
     ) == listeners_without_writes(init_listeners)
+
+
+async def test_history_stream_for_invalid_entity_ids(
+    recorder_mock: Recorder, hass: HomeAssistant, hass_ws_client: WebSocketGenerator
+) -> None:
+    """Test history stream for invalid and valid entity ids."""
+
+    now = dt_util.utcnow()
+    await async_setup_component(
+        hass,
+        "history",
+        {history.DOMAIN: {}},
+    )
+
+    await async_setup_component(hass, "sensor", {})
+    await async_recorder_block_till_done(hass)
+    hass.states.async_set("sensor.one", "on", attributes={"any": "attr"})
+    await async_recorder_block_till_done(hass)
+    hass.states.async_set("sensor.two", "off", attributes={"any": "attr"})
+    await async_recorder_block_till_done(hass)
+    hass.states.async_set("switch.excluded", "off", attributes={"any": "again"})
+    await async_wait_recording_done(hass)
+
+    await async_wait_recording_done(hass)
+
+    client = await hass_ws_client()
+
+    await client.send_json(
+        {
+            "id": 1,
+            "type": "history/stream",
+            "start_time": now.isoformat(),
+            "include_start_time_state": True,
+            "significant_changes_only": False,
+            "no_attributes": True,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    response = await client.receive_json()
+    assert "sensor.one" in str(response)
+    assert "sensor.two" in str(response)
+
+    await client.send_json(
+        {
+            "id": 2,
+            "type": "history/stream",
+            "start_time": now.isoformat(),
+            "entity_ids": ["sensor.one"],
+            "include_start_time_state": True,
+            "significant_changes_only": False,
+            "no_attributes": True,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    response = await client.receive_json()
+    assert "sensor.one" in str(response)
```

Comment:
Since the response is structured data we should assert its in the right place in the response instead of stringifying the response